packages/batch-submitter/src/batch-submitter/tx-batch-submitter.ts

```diff
@@ -432,6 +439,57 @@ export class TransactionBatchSubmitter extends BatchSubmitter {
       return fixedBatch
     }
 
+    const fixSkippedDeposits = async (b: Batch): Promise<Batch> => {
+      this.logger.debug('Fixing skipped deposits...')
+      let nextQueueIndex = await this.chainContract.getNextQueueIndex()
+      const fixedBatch: Batch = []
+      for (const ele of b) {
+        // Look for skipped deposits
+        while (true) {
+          const pendingQueueElements = await this.chainContract.getNumPendingQueueElements()
+          const nextRemoteQueueElements = await this.chainContract.getNextQueueIndex()
+          const totalQueueElements =
+            pendingQueueElements + nextRemoteQueueElements
```

Comment:
Was thinking this could be moved outside of the for loop, but I guess a new queue element could appear while this is running....

Reply:
You're right -- good catch lol. Ugh.... 😅